ert3/data/_record.py

```diff
@@ -163,66 +187,69 @@ def __init__(self, name: str, storage_path: Path):
-                async with aiofiles.open(storage_uri, mode="wb") as f:  # type: ignore
-                    await f.write(record.data[0])  # type: ignore
-            else:
-                raise TypeError(f"unexpected record data type {type(record.data)}")
-        self.set_transmitted(storage_uri, mime)
+            async with aiofiles.open(storage_uri, mode="wb") as f:  # type: ignore
+                await f.write(record.data[0])  # type: ignore
+        self.set_transmitted(storage_uri, record_type=record.record_type)
 
     async def transmit_data(
         self,
         data: Union[List[float], Mapping[int, float], Mapping[str, float], List[bytes]],
-        mime,
     ) -> None:
         if self.is_transmitted():
             raise RuntimeError("Record already transmitted")
         record = Record(data=data)
-        return await self._transmit(record, mime)
+        return await self._transmit(record)
 
     async def transmit_file(
         self,
         file: Path,
         mime,
     ) -> None:
         if self.is_transmitted():
-            raise RuntimeError("File already transmitted")
-        if mime != "application/json":
+            raise RuntimeError("Record already transmitted")
+        if mime == "application/json":
+            async with aiofiles.open(str(file), mode="r") as f:
+                contents = await f.read()
+                record = Record(data=json.loads(contents))
+        elif mime == "application/octet-stream":
+            async with aiofiles.open(str(file), mode="rb") as f:  # type: ignore
+                contents = await f.read()
+                record = Record(data=[contents])
+        else:
             raise NotImplementedError(
-                f"cannot transmit file unless json, was {self._mime}"
+                "cannot transmit file unless mime is application/json"
+                f" or application/octet-stream, was {mime}"
             )
-        async with aiofiles.open(str(file), mode="r") as f:
-            contents = await f.read()
-            record = Record(data=json.loads(contents))
-        return await self._transmit(record, mime)
+        return await self._transmit(record)
 
     async def load(self) -> Record:
-        if self._state != RecordTransmitterState.transmitted:
+        if not self.is_transmitted():
             raise RuntimeError("cannot load untransmitted record")
-        if self._mime != "application/json":
-            raise NotImplementedError(
-                f"cannot load record unless json, was {self._mime}"
-            )
-        async with aiofiles.open(str(self._uri)) as f:
-            contents = await f.read()
-            return Record(data=json.loads(contents))
+        if self._record_type != RecordType.LIST_BYTES:
+            async with aiofiles.open(str(self._uri)) as f:
+                contents = await f.read()
+                return Record(data=json.loads(contents))
```

Comment:
When did all records become json? Am confused

Reply:
The records are not json, but the SharedStorageTransmitter chooses to store them as json?